frontend/apps/app/app/api/chat/route.ts

```diff
@@ -0,0 +1,204 @@
+  })
+}
+
+// Convert schema data to text chunks for vector storage
+const convertSchemaToTexts = (schema: SchemaData): Document[] => {
+  const documents: Document[] = []
+
+  // Process tables
+  if (schema.tables) {
+    for (const [tableName, tableData] of Object.entries(schema.tables)) {
+      documents.push(tableToDocument(tableName, tableData))
+    }
+  }
+
+  // Process relationships
+  if (schema.relationships) {
+    for (const [relationshipName, relationshipData] of Object.entries(
+      schema.relationships,
+    )) {
+      documents.push(relationshipToDocument(relationshipName, relationshipData))
+    }
+  }
+
+  return documents
+}
+
+// Create vector store from schema data
+const createVectorStore = async (schemaData: SchemaData) => {
+  try {
+    const documents = convertSchemaToTexts(schemaData)
+    return await MemoryVectorStore.fromDocuments(
+      documents,
+      new OpenAIEmbeddings(),
+    )
+  } catch (error) {
+    Sentry.captureException(error)
+    console.error('Error creating vector store:', error)
+    throw error
+  }
+}
+
+// Create chat chain
+const createChatChain = async (vectorStore: MemoryVectorStore) => {
+  const model = new ChatOpenAI({
+    modelName: 'gpt-4',
+    temperature: 0,
+  })
+
+  // Create a prompt template
+  const prompt = ChatPromptTemplate.fromTemplate(`
+You are a database schema expert.
+Answer questions about the user's schema and provide advice on database design.
+Follow these guidelines:
+
+1. Clearly explain the structure of the schema, tables, and relationships.
+2. Provide advice based on good database design principles.
+3. Share best practices for normalization, indexing, and performance.
+4. When using technical terms, include brief explanations.
+5. Provide only information directly related to the question, avoiding unnecessary details.
+
+Your goal is to help users understand and optimize their database schemas.
+
+Context information:
+{context}
+
+Question: {input}
+
+Based on the context information, provide a helpful answer to the question.
+`)
+
+  // Create a document chain
+  const documentChain = await createStuffDocumentsChain({
+    llm: model,
+    prompt,
+  })
+
+  // Create the retrieval chain
+  return createRetrievalChain({
+    retriever: vectorStore.asRetriever(),
+    combineDocsChain: documentChain,
+  })
+}
+
+export async function POST(request: Request) {
+  try {
+    const { message, schemaData, history } = await request.json()
+
+    // Create vector store and chain
+    const vectorStore = await createVectorStore(schemaData)
+    const chain = await createChatChain(vectorStore)
+
+    // Format chat history
+    const formattedHistory = history
+      ? history.map((msg: [string, string]) =>
+          msg[0] === 'Human' ? new HumanMessage(msg[1]) : new AIMessage(msg[1]),
+        )
+      : []
+
+    // Generate response
+    const response = await chain.invoke({
+      input: message,
+      chat_history: formattedHistory,
+    })
```

Comment:
The history is not persistent and is temporarily stored in front-end memory (React state).